### PR TITLE
Add EIP-5554: NFT Legal Use, Sharing, Repurposing, and Remixing

### PR DESCRIPTION
**Pull request by ipatka**
A standard interface and licensing scheme for describing and enforcing the use, sharing, repurposing, and remixing rights associated with an NFT. Establishing an on-chain registry of IP rights associated with NFTs, and the corresponding attribution and derivative links.

### CODE REVIEW

**Comment by Pandapip1** [comment]
Please fix the rest of the CI errors. I've done the minimum amount of changes to fix most of the errors.

File: EIPS/eip-5554.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+Every contract compliant with this EIP must implement the `IERC5554` interface:
+
+```solidity
+pragma solidity ^0.8.0;
+
+interface IERC5554 is IERC5218 {
+
+    event CommercialExploitation(uint256 _tokenId, uint256 _licenseId, string _externalUri);
+    event ReproductionCreated(uint256 _tokenId, uint256 _licenseId, address _reproduction);
+    event DerivativeCreated(uint256 _tokenId, uint256 _licenseId, address _derivative);
```
Comment (iseriohn) [inline EIPS/eip-5554.md:49]:
```suggestion
    event ReproductionCreated(uint256 _tokenId, uint256 _licenseId, uint256 _reproductionId, address _reproduction, uint256 _reproductionTokenId);
    event DerivativeCreated(uint256 _tokenId, uint256 _licenseId, uint256 _derivativeId, address _derivative, uint256 _derivativeTokenId);
```

1. If `_reproductionId` and `_derivativeId` is assigned at execution, it's better to log them in events as well for easy filtering and tracking;
2. The reproductions and derivatives may not exist as an individual collection but in collections of other works as well, e.g., consider the Opensea smart contract. So it's better to log the token ID for them in addition to the smart contract address.

File: EIPS/eip-5554.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    function getCopyrightOwner(uint256 tokenId)
+        external
+        virtual
+        returns (address);
+    
+    /// @notice Requests to log an execution of a license
+    /// @dev Throws unless the token issuance conditions are met
+    /// @param tokenId The identifier for the queried token
+    /// @return uint256 tracking reproduction ID
+    function logReproduction(uint256 tokenId, address reproduction)
```
Comment (iseriohn) [inline EIPS/eip-5554.md:64]:
```suggestion
    function logReproduction(uint256 tokenId, address reproduction, uint256 reproductionTokenId)
```

File: EIPS/eip-5554.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        external
+        virtual
+        returns (uint256);
+
+    /// @notice Requests to log an executions of a license
+    /// @dev Throws unless the token issuance conditions are met
+    /// @param tokenId The identifier for the queried token
+    /// @return uint256 tracking derivative ID
+    function logDerivative(uint256 tokenId, address derivative)
```
Comment (iseriohn) [inline EIPS/eip-5554.md:73]:
```suggestion
    function logDerivative(uint256 tokenId, address derivative, uint256 derivativeTokenId)
```

**Comment by ipatka** [comment]
thanks @iseriohn! integrated all of the changes

**Review by Pandapip1 — APPROVED** [review]
There's some stuff I would like fixed before this moves in to review, but this is fine for a draft.

**Pull request merged**